docs: add windows tab for using Shorebird on Codemagic Windows machines (#108)
shorebirdtech/docs: docs/ci/codemagic.mdx

  </TabItem>
  <TabItem value="windows" label="Windows" default>

1. Add the group "shorebird" to your environment groups. If you're using a

>    different group name, use that instead. Also add a variable named `shorebird`

Reviewer: Codemagic noob question: when/why would I be using "shorebird" as my group name vs a different group name?

Author: In Codemagic, you group single secrets into a secret group. Imagine you have the following secrets to upload your iOS app to App Store Connect:

* APP_STORE_CONNECT_PRIVATE_KEY
* APP_STORE_CONNECT_ISSUER_ID
* APP_STORE_CONNECT_KEY_IDENTIFIER 

Now, you bundle them in a group like "ios-publishing". Instead of listing every single secret in the `codemagic.yaml` file, you can just list the group "ios-publishing" and this will add the three secrets.

It's not possible to just a single secret (like in GitHub Actions). A secret always needs a group.

Back to your question: You would use "shorebird" as your group when only have related secrets to Shorebird in this group. However, could also add the `SHOREBIRD_TOKEN` to a different group. In this case, you would need to replace `shorebird` as group. 

Here is an example screenshot from the Codemagic console:

![image](https://github.com/shorebirdtech/docs/assets/24459435/82f876e2-b0a7-40fe-98c4-a93444a59f47)

> 1. Add the group "shorebird" to your environment groups. If you're using a
>    different group name, use that instead. Also add a variable named `shorebird`

Reviewer: ```suggestion
1. Add the group containing `SHOREBIRD_TOKEN` to your environment groups. Also add a variable named `shorebird`
```

Reviewer: Makes sense. Minor edit for clarity (if I'm understanding correctly).

Author: As context: In the section above, I used "shorebird" as group name. Should we also mention the group name "shorebird"?

![image](https://github.com/shorebirdtech/docs/assets/24459435/558c35d4-0405-47fe-99f0-774cd6f16923)

Author: Yes, that sounds great! Thanks for the suggestion!

   with the path to the Shorebird executable.

```yaml
workflows:
  example:
    name: Example
    environment:
      groups:
        # Exports the SHOREBIRD_TOKEN environment variable
        - shorebird
      flutter: stable
      vars:
        # Workaround for
        # https://github.com/orgs/codemagic-ci-cd/discussions/1921#discussioncomment-6582318
        shorebird: "C:\\Users\\builder\\.shorebird\\bin\\shorebird.ps1"
```

2. Add a script to your workflow to set up Shorebird. This script will install
   Shorebird.

```yaml
scripts:
  - name: 🐦 Setup Shorebird
    script: |
      Set-ExecutionPolicy RemoteSigned -scope CurrentUser # Needed to execute remote scripts
      iwr -UseBasicParsing 'https://raw.githubusercontent.com/shorebirdtech/install/main/install.ps1'|iex
```

3. Run Shorebird commands with `& $env:shorebird` instead of just `shorebird` as a workaround for [codemagic-ci-cd/discussions/1921](https://github.com/orgs/codemagic-ci-cd/discussions/1921#discussioncomment-6582318).

```yaml
scripts:
  - name: 🚀 Shorebird Patch
    script: |

>       & $env:shorebird patch android --force

Reviewer 2: ```suggestion
      & $env:shorebird patch android --release-version <RELEASE-VERSION> --force
```

The latest version requires users to specify which release version they are patching in order for shorebird to know which revision of Flutter to use.

Author: Ah, good to know. I created #110 to adjust the "Advanced example". For the windows tab I'm going to show the easy case, so I replaced `patch` with `release`.
